src/matcher/plugin.py

```diff
@@ -336,4 +383,19 @@ def pytest_addoption(parser) -> None:
       , default=False
       )
 
+
+@pytest.hookimpl(trylast=True)
+def pytest_configure(config: pytest.Config):
+    """Register additional configuration."""
+    if os.getenv('PYTEST_MATCHER_RETURN_CODES', '').lower() == 'yes':  # NOQA: SIM108
+        plugin_return_codes = True
+    else:
+        plugin_return_codes = False
```

Comment:
Why not simple like this?
```suggestion
    plugin_return_codes = os.getenv('PYTEST_MATCHER_RETURN_CODES', '').lower() == 'yes'
```

Reply:
Oe even like this
```suggestion
    plugin_return_codes = os.getenv('PYTEST_MATCHER_RETURN_CODES', '').lower() in ('yes', 'true', '1')
```

Reply:
Except for this question, the rest is LGTM.